packages/medusa/src/types/global.ts

```diff
@@ -71,7 +77,7 @@ export type ConfigModule = {
     admin_cors?: string
   }
   featureFlags: Record<string, boolean | string>
-  modules?: Record<string, string>
+  modules?: Record<string, boolean | string | ConfigurableModuleDeclaration>
```

Comment:
nit: we will never allow `true` so we could change to:
```suggestion
  modules?: Record<string, false | string | ConfigurableModuleDeclaration>
```